schemes/musica/tuvx/musica_ccpp_tuvx_gas_species_profile.F90

```diff
@@ -0,0 +1,219 @@
+    type(profile_t), pointer :: profile
+  end type profile_group_t
+
+contains
+  !> Creates 'gas_species_t' object
+  function create_gas_species(label, unit, molar_mass, scale_height, &
+      index_constituent_props) result( gas_species )
+
+    character(len=*), intent(in) :: label
+    character(len=*), intent(in) :: unit
+    real(kind_phys),  intent(in) :: molar_mass ! kg mol-1
+    real(kind_phys),  intent(in) :: scale_height ! km
+    integer,          intent(in) :: index_constituent_props
+    type(gas_species_t)          :: gas_species
+
+    gas_species%label = label
+    gas_species%unit = unit
+    gas_species%molar_mass = molar_mass
+    gas_species%scale_height = scale_height
+    gas_species%index_constituent_props = index_constituent_props
+
+  end function create_gas_species
+
+  !> Configures gas species.
+  ! Note: The user can customize this function to configure gas species in the system
+  !       This function allocates memory for the gas species that is needed in tuvx_init and tuvx_run. 
+  !       The user is responsible for deallocating the memory.
+  subroutine configure_gas_species(constituent_props, gas_species_group, errmsg, errcode)
+    use ccpp_const_utils,          only: ccpp_const_get_idx
+    use ccpp_constituent_prop_mod, only: ccpp_constituent_prop_ptr_t
+
+    type(ccpp_constituent_prop_ptr_t), intent(in)  :: constituent_props(:)
+    type(gas_species_t), allocatable,  intent(out) :: gas_species_group(:)
+    character(len=512),                intent(out) :: errmsg
+    integer,                           intent(out) :: errcode
+
+    ! local variable
+    integer                    :: num_gas_species = 3
+    real(kind_phys), parameter :: SCALE_HEIGHT_DRY_AIR = 8.01_kind_phys    ! km
+    real(kind_phys), parameter :: SCALE_HEIGHT_O2 = 7.0_kind_phys          ! km
+    real(kind_phys), parameter :: SCALE_HEIGHT_O3 = 7.0_kind_phys          ! km
+    real(kind_phys)            :: molar_mass_O2                            ! kg mol-1
+    real(kind_phys)            :: molar_mass_O3                            ! kg mol-1
+    integer,         parameter :: INDEX_NOT_KNOWN = -9999
+    integer                    :: index_air, index_O2, index_O3
+
+    allocate(gas_species_group(num_gas_species)) ! need to deallocate 
+  
+    ! TODO(jiwon) - I commented out this block of code that searches for the molar mass
+    ! of air and instead hard-coded it using the value imported from CAM-SIMA.
+    ! Should we register air in the register phase like cloud liquid water content,
+    ! or should we include it in the configuration file, or is it good as it is?
+  
+    ! Air
+    ! Find the index of the species and create a 'gas_species_t' for it
+    ! call ccpp_const_get_idx(constituent_props, "air", index_air, errmsg, errcode)
+    ! write(*,*) "air index: ", index_air
+    ! if (errcode /= 0) return
+
+    ! call constituent_props(index_air)%molar_mass(molar_mass_air, errcode, errmsg)
+    ! if (errcode /= 0) return
+
```

Comment:
Hi @mattldawson, I was trying to make the gas species configurable, allowing the user to provide their own species and define a function. Do you have any feedback on this implementation? Also, `air` is not included in the `species.json`, so it ends up being hard-coded. Could you share your thoughts on that in my comment?